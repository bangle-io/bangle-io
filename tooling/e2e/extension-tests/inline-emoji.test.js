const {
  url,
  createNewNote,
  sendCtrlABackspace,
  getEditorHTML,
  createWorkspace,
  sleep,
  longSleep,
  getPrimaryEditorDebugString,
} = require('../helpers');
jest.setTimeout(105 * 1000);

beforeEach(async () => {
  await jestPuppeteer.resetPage();
  await page.goto(url, { waitUntil: 'networkidle2' });

  await page.evaluate(() => localStorage.clear());
});

test('Emoji works in heading', async () => {
  const wsName = await createWorkspace(page);

  await createNewNote(page, wsName, 'test123');
  await longSleep();

  const editorHandle = await page.$('.bangle-editor');
  await sendCtrlABackspace(page);
  await sleep();
  await editorHandle.type('# Wow :', { delay: 3 });
  await editorHandle.press('ArrowDown');
  await editorHandle.press('Enter');

  const html = await getEditorHTML(editorHandle);

  expect(html.includes('ðŸ˜‰')).toBe(true);

  expect(await getPrimaryEditorDebugString(page)).toMatchSnapshot();
});

test('Emoji works in para', async () => {
  const wsName = await createWorkspace(page);

  await createNewNote(page, wsName, 'test123');

  const editorHandle = await page.$('.bangle-editor');

  await sendCtrlABackspace(page);

  await editorHandle.type('life is good :zeb', { delay: 1 });
  await editorHandle.press('Enter');
  const html = await getEditorHTML(editorHandle);
  expect(html.includes('ðŸ¦“')).toBe(true);
  expect(await getEditorHTML(editorHandle)).toMatchSnapshot();
});

test('Emoji works in list', async () => {
  const wsName = await createWorkspace(page);

  await createNewNote(page, wsName, 'test123');

  const editorHandle = await page.$('.bangle-editor');

  await sendCtrlABackspace(page);

  await editorHandle.type('- I am a list :zeb', { delay: 1 });
  await editorHandle.press('Enter');
  const html = await getEditorHTML(editorHandle);
  expect(html.includes('ðŸ¦“')).toBe(true);
  expect(await getEditorHTML(editorHandle)).toMatchSnapshot();
});
