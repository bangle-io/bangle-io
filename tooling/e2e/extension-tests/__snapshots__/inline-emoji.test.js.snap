// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emoji works in heading 1`] = `"doc(heading(\\"Wow \\", emoji, \\" \\"), paragraph)"`;

exports[`Emoji works in list 1`] = `
"<ul>
  <li
    data-bangle-name=\\"listItem\\"
    class=\\"bangle-nv-container\\"
  >
    <span class=\\"bangle-nv-content\\"
      ><p class=\\"selected-node\\">
        I am a list
        <span
          data-bangle-name=\\"emoji\\"
          data-bangle-attrs='{\\"emojiAlias\\":\\"zebra\\"}'
          contenteditable=\\"false\\"
          draggable=\\"true\\"
          >ðŸ¦“</span
        >
      </p></span
    >
  </li>
</ul>
<p>
  <br
    class=\\"
      ProseMirror-trailingBreak
    \\"
  />
</p>
"
`;

exports[`Emoji works in para 1`] = `
"<p class=\\"selected-node\\">
  life is good
  <span
    data-bangle-name=\\"emoji\\"
    data-bangle-attrs='{\\"emojiAlias\\":\\"zebra\\"}'
    contenteditable=\\"false\\"
    draggable=\\"true\\"
    >ðŸ¦“</span
  >
</p>
"
`;
